src/sdam/monitor.rs

```diff
@@ -184,7 +184,12 @@ impl HeartbeatMonitor {
         let result = match self.connection {
             Some(ref mut conn) => {
                 let command = is_master_command(self.client_options.server_api.as_ref());
-                run_is_master(command, conn).await
+                // SDAM events should only be emitted if the topology is still alive
```

Comment:
This will still be racy since the topology could become non alive while `run_is_master` is executing, so if we have tests relying on this behavior we may need to still have special handling.